core/org.openjdk.jmc.common/src/main/java/org/openjdk/jmc/common/io/IOToolkit.java

```diff
@@ -147,30 +147,27 @@ public static InputStream openUncompressedStream(File file) throws IOException {
 	 *             on I/O error
 	 */
 	public static InputStream openUncompressedStream(InputStream stream) throws IOException {
-		InputStream in = stream;
-		if (in.markSupported()) {
-			in.mark(MAGIC_GZ.length + 1);
-			if (hasMagic(in, MAGIC_GZ)) {
-				in.reset();
-				return new GZIPInputStream(in);
-			}
+		InputStream in = new BufferedInputStream(stream);
+		in.mark(MAGIC_GZ.length + 1);
+		if (hasMagic(in, MAGIC_GZ)) {
 			in.reset();
-			in.mark(MAGIC_ZIP.length + 1);
-			if (hasMagic(in, MAGIC_ZIP)) {
-				in.reset();
-				ZipInputStream zin = new ZipInputStream(in);
-				zin.getNextEntry();
-				return zin;
-			}
+			return new GZIPInputStream(in);
+		}
+		in.reset();
+		in.mark(MAGIC_ZIP.length + 1);
+		if (hasMagic(in, MAGIC_ZIP)) {
 			in.reset();
-			in.mark(MAGIC_LZ4.length + 1);
-			if (hasMagic(in, MAGIC_LZ4)) {
-				in.reset();
-				return new LZ4FrameInputStream(in);
-			}
+			ZipInputStream zin = new ZipInputStream(in);
+			zin.getNextEntry();
+			return zin;
+		}
+		in.reset();
+		in.mark(MAGIC_LZ4.length + 1);
+		if (hasMagic(in, MAGIC_LZ4)) {
 			in.reset();
+			return new LZ4FrameInputStream(in);
 		}
-		in = new BufferedInputStream(stream);
+		in.reset();
 		return in;
```

Comment:
This diff is much easier to follow with whitespace hidden. By moving the `BufferedInputStream` to the beginning of the method we can guarantee mark is supported and remove the conditional.